src/pages/execution/workflowTabs/gantt-chart/atoms.tsx

```diff
@@ -0,0 +1,169 @@
+import { atom, useAtomValue } from 'jotai';
+import { getTextWidth, fontFamily, fontSizes } from './internal/utils';
+import { idAccessor, yAccessor } from './utils';
+import { scaleBand, scaleOrdinal, scaleTime, schemeGreens } from 'd3';
+import { useAtomSetterOnChange } from './internal/hooks';
+import React from 'react';
+import isEqual from 'lodash/isEqual';
+import type { Band, Margins, Series } from './types';
+
+// MARK - html elements
+
+export const canvasAtom = atom<HTMLCanvasElement>(null as HTMLCanvasElement);
+canvasAtom.onMount = (setAtom) => {
+    const canvas = document.createElement('canvas');
+    setAtom(canvas);
+    return () => canvas.remove();
+};
+
+// MARK - props data atoms
+
+export const maxAtom = atom(new Date());
+
+export const minAtom = atom(new Date(0));
+
+export const bandAtom = atom<Band>({
+    height: 45,
+    padding: 15,
+});
+
+export const marginsAtom = atom<Margins>({
+    top: 0,
+    left: 20,
+    right: 50,
+    bottom: 50,
+});
+
+export const rowsAtom = atom<Pick<Series, 'id' | 'label'>[]>([]);
+
+export const scopedMinMaxAtom = atom<[Date, Date]>([new Date(0), new Date()]);
+
+export const barIdsAtom = atom<string[]>([]);
+
+// MARK - graph measurements atoms
+
+export const graphWidthAtom = atom(0, null);
+
+export const yAxisWidthAtom = atom((get) => {
```

Comment:
looks like you've stripped out use of `yAxisWidthAtom` already. Can remove completely.